src/main/java/org/broadinstitute/hellbender/tools/copynumber/CollectFragmentCounts.java

```diff
@@ -0,0 +1,223 @@
+        filters.add(ReadFilterLibrary.MAPPED);
+        filters.add(ReadFilterLibrary.NON_ZERO_REFERENCE_LENGTH_ALIGNMENT);
+        filters.add(ReadFilterLibrary.NOT_DUPLICATE);
+        filters.add(ReadFilterLibrary.FIRST_OF_PAIR); // this will make sure we don't double count
+        filters.add(ReadFilterLibrary.PROPERLY_PAIRED);
+        // this will only keep reads in pairs that are properly oriented and mapped on same chromosome
+        // and lie within a few standard deviations from the mean of fragment size distributions
+        return filters;
+    }
+
+    @Override
+    public boolean requiresIntervals() {
+        return true;
+    }
+
+    @Override
+    public void onTraversalStart() {
+        if (readArguments.getReadFilesNames().size() != 1) {
+            throw new UserException("This tool only accepts a single bam/sam/cram as input");
+        }
+
+        final SampleCollection sampleCollection = new SampleCollection(getHeaderForReads());
+        if (sampleCollection.sampleCount() > 1) {
+            throw new UserException.BadInput("We do not support bams with more than one sample.");
+        }
+        sampleName = sampleCollection.sampleIds().get(0);
+
+        final List<SimpleInterval> intervals = intervalArgumentCollection.getIntervals(sequenceDictionary);
+        cachedIntervalList = new CachedBinarySearchIntervalList<>(intervals);
+        intervalCounts = new int[intervals.size()];
+
+        logger.info("Starting to collect fragment counts...");
+    }
+
+    @Override
+    public void apply(GATKRead read, ReferenceContext referenceContext, FeatureContext featureContext) {
+        //getting a center of the fragment
+
+        //TODO collect information on reads that do not have a properly paired mate
+        int centerOfFragment = ReadOrientation.getCenterOfFragment(read);
+        final Locatable centerFragmentLocation = new SimpleInterval(read.getContig(), centerOfFragment, centerOfFragment);
+
+        final IndexRange intersectionRange = cachedIntervalList.findIntersectionRange(centerFragmentLocation);
+        if (intersectionRange.size() > 1) {
+            // should not reach here since intervals are checked for overlapping;
+            // doing a check to protect against future code changes
+            throw new GATKException.ShouldNeverReachHereException("At most one interval can intersect with a center of a fragment.");
+        }
+        intersectionRange.forEach(intervalIndex -> intervalCounts[intervalIndex]++);
+    }
+
+    @Override
+    public Object onTraversalSuccess() {
+        logger.info("Finished collecting fragment files");
+        logger.info("Writing results to disk");
+        writeResults(outputCountsFile, sampleName);
+
+        logger.info("Fragment counts written to " + outputCountsFile.toString());
+        return "SUCCESS";
+    }
+
+    private void writeResults(final File outputFile, final String sampleName) {
```

Comment:
@mbabadi Do you need these files to use the old target format?  If not, let's go ahead and switch over to something more reasonable.  If so, let's sequester this code in a copynumber/temporary package.